integration_test/general.py

```diff
@@ -0,0 +1,154 @@
+    return runCommand("./integration_test/scripts/run-import-adapter-http2.sh "+adapter_port+" "+version+" "+project+" "+location+" "+dataset+" "+store_name, "run import adapter exit with")
+
+# run import adapter with local backup mode
+def run_import_adapter_local_backup(adapter_port, version, project, location, dataset, store_name):
+    return runCommand("./integration_test/scripts/run-import-adapter-local-backup.sh "+adapter_port+" "+version+" "+project+" "+location+" "+dataset+" "+store_name, "run import adapter exit with")
+
+# run import adapter with gsc backup mode
+def run_import_adapter_gcs_backup(adapter_port, version, project, location, dataset, store_name, bucket):
+    return runCommand("./integration_test/scripts/run-import-adapter-gcs-backup.sh "+adapter_port+" "+version+" "+project+" "+location+" "+dataset+" "+store_name+" "+bucket, "run import adapter exit with")
+
+# wait-for-adapter
+# wait-for-storescp
+def wait_for_port(host, port):
+    return runCommand("./integration_test/scripts/wait-for-port.sh "+host+" " + port, "wait-for-adapter exit with")
+
+# run-store-scu
+def run_store_scu(host, adapter_port, file_path):
+    return runCommand("export PATH=/opt/apache-maven-3.6.3/bin:$PATH &&"
+           "cd dcm4che/dcm4che-tool/dcm4che-tool-storescu &&"
+           "mvn -ntp exec:java -Dexec.mainClass=org.dcm4che3.tool.storescu.StoreSCU -Dexec.args=-\"c IMPORTADAPTER@"+host+":"+adapter_port+" "+file_path+"\"", "run-store-scu exit with")
+
+# run-find-scu-instance
+def run_find_scu_instance(host, adapter_port):
+    return runCommand("export PATH=/opt/apache-maven-3.6.3/bin:$PATH &&"
+           "cd /workspace/dcm4che/dcm4che-tool/dcm4che-tool-findscu &&"
+           "mvn -ntp  exec:java -Dexec.mainClass=org.dcm4che3.tool.findscu.FindSCU -Dexec.args=\"-c IMPORTADAPTER@"+host+":"+adapter_port+" -L IMAGE -X --out-cat --out-file findscu-instance-result.xml --out-dir ../../../integration_test/\"", "run-find-scu-instance exit with")
+
+# run-find-scu-series
+def run_find_scu_series(host, adapter_port):
+    return runCommand("export PATH=/opt/apache-maven-3.6.3/bin:$PATH &&"
+           "cd /workspace/dcm4che/dcm4che-tool/dcm4che-tool-findscu &&"
+           "mvn -ntp  exec:java -Dexec.mainClass=org.dcm4che3.tool.findscu.FindSCU -Dexec.args=\"-c IMPORTADAPTER@"+host+":"+adapter_port+" -L SERIES -X --out-cat --out-file findscu-series-result.xml --out-dir ../../../integration_test/\"", "run-find-scu-series exit with")
+
+# run-find-scu-study
+def run_find_scu_study(host, adapter_port):
+    return runCommand("export PATH=/opt/apache-maven-3.6.3/bin:$PATH &&"
+           "cd /workspace/dcm4che/dcm4che-tool/dcm4che-tool-findscu &&"
+           "mvn -ntp  exec:java -Dexec.mainClass=org.dcm4che3.tool.findscu.FindSCU -Dexec.args=\"-c IMPORTADAPTER@"+host+":"+adapter_port+" -L STUDY -X --out-cat --out-file findscu-study-result.xml --out-dir ../../../integration_test/\"", "run-find-scu-study exit with")
```

Comment:
These names "findscu-study-result.xml", "findscu-series-result.xml", "findscu-instance-result.xml" do not match the file names in the test itself "findscu-instance-result1.xml"

Shouldn't the diff be failing in that case?

Reply:
We handle different files for each case :
verify_result(run_find_scu_instance(substitution.STORE_SCP_RUN_STEP, substitution.ADAPTER_PORT))
verify_result(run_find_scu_series(substitution.STORE_SCP_RUN_STEP, substitution.ADAPTER_PORT))
verify_result(run_find_scu_study(substitution.STORE_SCP_RUN_STEP, substitution.ADAPTER_PORT))

And we compare the result for each case :
verify_result(check_diff("integration_test/findscu-instance-result1.xml", "integration_test/data/findscu-instance-expected.xml"))
verify_result(check_diff("integration_test/findscu-series-result1.xml", "integration_test/data/findscu-series-expected.xml"))
verify_result(check_diff("integration_test/findscu-study-result1.xml", "integration_test/data/findscu-study-expected.xml"))

Reply:
What I mean is, the --out-file name "findscu-study-result.xml" is different from the check_diff arg "findscu-study-result1.xml"

Reply:
I took these arguments from the current version of the tests:

id: run-find-scu-study
   - '-Dexec.args=-c IMPORTADAPTER@${_ADAPTER_RUN_STEP}:${_ADAPTER_PORT} -L STUDY -X --out-cat --out-file findscu-study-result.xml --out-dir ../../../integration_test/'
   
id: check-find-diff-study
  args:
  - diff
  - 'integration_test/findscu-study-result1.xml'
  - 'integration_test/data/findscu-study-expected.xml'

Reply:
I think it's how the tool writes the files. Thanks.